src/index.js

```diff
@@ -44,6 +45,41 @@ class Index extends Component {
     this.onFinish = this.onFinish.bind(this);
   }
 
+  componentDidMount() {
+      let url_parameter = {};
+      const currLocation = this.props.location.search;
+      let parArr = currLocation.split("?");
+      if (parArr.length > 1) {
+          parArr = parArr[1].split("&");
+          for (let i = 0; i < parArr.length; i++) {
+              const parr = parArr[i].split("=");
+              if (parr[0] === 'lang') {
```

Comment:
If the url contains 'lang', we need to update language string then exit the componentDidMount(), no need to check `window.localStorage.getItem('lang')`